relay/server.go

```diff
@@ -163,6 +162,45 @@ func NewServer(
 	}, nil
 }
 
+// NewInstanceWithDependencies creates and starts a relay server instance with the given dependencies.
+// This is a convenience function that combines NewServer + StartAsInstanceWithListener.
```

Comment:
There's too many similar convenience functions. I'd vote for just keeping NewServer. Let the dependencies be constructed and passed at call sites. We don't need NewInstanceWithDependencies or StartAsInstance imo. The listener should prob just be passed to the constructor, and then Start() would use it.